### PR TITLE
Add support for DHCP options set

### PR DESCRIPTION
**Pull request by bcenker**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -11,6 +11,28 @@ resource "aws_vpc" "this" {
 }
 
 ###################
+# DHCP Options Set
+###################
+resource "aws_vpc_dhcp_options" "this" {
+  count = "${ var.enable_dhcp_options ? 1 : 0 }"
```
Comment (antonbabenko) [inline main.tf]: 1. `count = "${var.enable_dhcp_options ? 1 : 0 }"`
1. Add new line after `count` argument

File: main.tf [hunk 2]
```diff
@@ -11,6 +11,28 @@ resource "aws_vpc" "this" {
...
+  domain_name = "${var.dhcp_options_domain_name}"
+  domain_name_servers = "${var.dhcp_options_dns_servers}"
```
Comment (antonbabenko) [inline main.tf]: Rename dhcp_options_dns_servers to dhcp_options_domain_name_servers

File: main.tf [hunk 3]
```diff
@@ -11,6 +11,28 @@ resource "aws_vpc" "this" {
...
+  ntp_servers = "${var.dhcp_options_ntp_servers}"
+  netbios_name_servers = "${var.dhcp_options_netbios_servers}"
```
Comment (antonbabenko) [inline main.tf]: Rename dhcp_options_netbios_servers to dhcp_options_netbios_name_servers

File: main.tf [hunk 4]
```diff
@@ -11,6 +11,28 @@ resource "aws_vpc" "this" {
...
+  netbios_node_type = "${var.dhcp_options_netbios_node_type}"
+}
+
+###############################
+# DHCP Options Set Association
+###############################
+resource "aws_vpc_dhcp_options_association" "this" {
+  count = "${ var.enable_dhcp_options ? 1 : 0 }"
```
Comment (antonbabenko) [inline main.tf]: Same as for previous `count`

File: variables.tf [hunk 5]
```diff
@@ -124,3 +124,36 @@ variable "elasticache_subnet_tags" {
...
+}
+
+variable "dhcp_options_domain_name" {
+  description = "Specifies DNS name for DHCP options set"
+  default     = ""
+}
+
+variable "dhcp_options_dns_servers" {
+  type        = "list"
+  description = "Specify a list of DNS server addresses for DHCP options set, default to AWS provided"
```
Comment (antonbabenko) [inline variables.tf]: Reorder a bit - description, type, default

**Comment by bcenker** [comment]
Hi Anton - Hopefully I addressed all the requested changes - thanks!

File: main.tf [hunk 6]
```diff
@@ -11,6 +11,30 @@ resource "aws_vpc" "this" {
...
+
+###############################
+# DHCP Options Set Association
+###############################
+resource "aws_vpc_dhcp_options_association" "this" {
+  count = "${var.enable_dhcp_options ? 1 : 0}"
+
+  vpc_id          = "${aws_vpc.this.id}"
+  dhcp_options_id = "${aws_vpc_dhcp_options.this.id}"
+  depends_on      = ["aws_vpc.this", "aws_vpc_dhcp_options.this"]
```
Comment (antonbabenko) [inline main.tf]: `depends_on` is not really needed here. Or, if this is this an edge case then provide an open issue link.

Comment (bcenker) [reply main.tf]: Agreed - no edge case that I know of, I'm not entirely sure why I put that in there.  Fixed!

**Comment by antonbabenko** [comment]
Brian, you are rock-star of the Saturday! [v1.2.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/tag/v1.2.0) is out :)

**Pull request merged**